app/src/main/java/ayds/winchester/songinfo/home/view/DateDescription/DatePrecisionText.kt

```diff
@@ -0,0 +1,60 @@
+import ayds.winchester.songinfo.home.model.entities.DatePrecision
+
+interface DateText{
+    fun getTextDate(date: String, precision: DatePrecision): String
+}
+
+object DateTextImpl : DateText{
+
+    override fun getTextDate(date: String, precision: DatePrecision): String {
+        var value_answer = ""
+        when (precision) {
+            DatePrecision.DAY -> value_answer = getYearMonthDay(date)
+            DatePrecision.MONTH -> value_answer = getYearMonth(date)
+            DatePrecision.YEAR -> value_answer = getYear(date)
+        }
+        return value_answer
+    }
+
+    private fun getYearMonthDay(date: String):String{
+        return date.replace("-", "/")
+    }
+
+    fun getYearMonth(date: String):String{
+        var new_year = date.substringBeforeLast("-")
+        var new_month = date.substringAfterLast("-")
+
+        return (getMonthName(new_month.toInt()) + ", " + new_year)
+    }
+
+    fun getYear(date: String):String{
+        var new_year = date.substringBeforeLast("-")
+        return (new_year +" "+ esBisiesto(new_year.toInt()))
+    }
+
+    fun getMonthName (month:Int):String{
+        val month_name = when (month) {
+            1 -> "January"
+            2 -> "February"
+            3 -> "March"
+            4 -> "April"
+            5 -> "May"
+            6 -> "June"
+            7 -> "July"
+            8 -> "August"
+            9 -> "September"
+            10 -> "October"
+            11 -> "November"
+            12 -> "December"
+            else -> {""}
+        }
+        return month_name;
+    }
+
+    fun esBisiesto(year: Int) : String{
```

Comment:
mejor en inglés, por uniformidad
```suggestion
    fun isLeapYear(year: Int) : String{
```